gpio/imx6ull-gpio/imx6ull-gpio.c

```diff
@@ -212,13 +208,35 @@ void thread(void *arg)
 				break;
 
 			case mtWrite:
-				if (msg.i.data != NULL && msg.i.size >= sizeof(uint32_t)) {
-					memcpy(&val, msg.i.data, sizeof(uint32_t));
+				if (msg.i.data != NULL && msg.i.size >= 2) {
+					uint8_t *data = (uint8_t *)msg.i.data;
+
+					/* NOTE: ignoring msg.i.io.offs to allow repetetive writes without seek / reopen */
+
+					if ((msg.i.size % sizeof(uint32_t) != 0) && ((data[0] == '+') || (data[0] == '-'))) {
+						printf("GPIO: text mode!\n");
+						/* text mode [+/-][pin] -> 1/0 as out/in or value */
+						uint32_t pin = 0;
```

Comment:
variable declaration not at the beginning of the scope

Reply:
shit, forgot to remove debug printf, will fix once @agkaminski would review.